stable/osg-hosted-ce/osg-hosted-ce/templates/deployment.yaml

```diff
@@ -23,9 +23,14 @@ spec:
         release: {{ .Release.Name }}
         instance: {{ .Values.Instance }}
     spec:
-      {{ if .Values.NodeSelection }}
+      {{ if .Values.NodeSelection.Enabled }}
       nodeSelector: 
-        kubernetes.io/hostname: {{ .Values.NodeSelection }}
+        {{ if .Values.NodeSelection.Hostname }}
+        kubernetes.io/hostname: {{ .Values.NodeSelection.Hostname }}
+        {{ end }}
+        {{ if .Values.NodeSelection.LabelSelectionFieldKey }}
+        {{ .Values.NodeSelection.LabelSelectionFieldKey }}: {{ .Values.NodeSelection.LabelSelectionFieldValue }} 
+        {{ end }}
```

Comment:
Do we only want to allow one field key/value pair? Otherwise, I would think that we could let operators provide a list of key/value pairs and loop over them

Reply:
How would you do this? Sorry, I don't know much about YAML. I believe in the original ticket the user only specified one key/value pair, but I agree having a list would be helpful.

Reply:
Check out how we do our SciToken mappings:

- https://github.com/slateci/slate-catalog/blob/master/stable/osg-hosted-ce/osg-hosted-ce/values.yaml#L81-L124
- https://github.com/slateci/slate-catalog/blob/master/stable/osg-hosted-ce/osg-hosted-ce/templates/configmap.yaml#L102-L117